File: src/data/pois.ts
import { POI } from '../types';

// Image-relative positions derived from the festival map image.
// x=0 is left edge, x=1 is right edge; y=0 is top, y=1 is bottom.
// These positions match the stylised Wildeburg map image.
export const POIS: POI[] = [
  // ── Stages & Venues ─────────────────────────────────────────────
  {
    id: 'wildlive',
    name: 'Wildlive',
    category: 'stage',
    imagePos: { x: 0.19, y: 0.32 },
    description: 'Main stage – live music',
  },
  {
    id: 'de-spot',
    name: 'De Spot',
    category: 'stage',
    imagePos: { x: 0.14, y: 0.21 },
    description: 'Electronic & DJ stage',
  },
  {
    id: 'radio-koperen-hond',
    name: 'Radio de Koperen Hond',
    category: 'stage',
    imagePos: { x: 0.43, y: 0.33 },
    description: 'Stage & radio broadcast',
  },
  {
    id: 'bud-kas',
    name: 'Bud Kas',
    category: 'stage',
    imagePos: { x: 0.50, y: 0.42 },
    description: 'Bar & stage',
  },
  {
    id: 'lod',
    name: 'LOD',
    category: 'stage',
    imagePos: { x: 0.59, y: 0.46 },
    description: 'Stage',
  },
  {
    id: 'bamboe-bios',
    name: 'Bamboe Bios',
    category: 'activity',
    imagePos: { x: 0.31, y: 0.67 },
    description: 'Outdoor cinema in the bamboo',
  },

  // ── Areas & Activities ───────────────────────────────────────────
  {
    id: 'strand',
    name: 'Strand',
    category: 'water',
    imagePos: { x: 0.12, y: 0.40 },
    description: 'Beach area',
  },
  {
    id: 'de-baan',
    name: 'De Baan',
    category: 'activity',
    imagePos: { x: 0.09, y: 0.46 },
    description: 'Activity track',
  },
  {
    id: 'bamboe-bos',
    name: 'Bamboe Bos',
    category: 'nature',
    imagePos: { x: 0.46, y: 0.52 },
    description: 'Bamboo forest',
  },
  {
    id: 'helling',
    name: 'Helling',
    category: 'nature',
    imagePos: { x: 0.12, y: 0.53 },
    description: 'The slope',
  },
  {
    id: 'duin-pan',
    name: 'Duin/Pan',
    category: 'nature',
    imagePos: { x: 0.36, y: 0.57 },
    description: 'Dune area',
  },
  {
    id: 'achter-lijn',
    name: 'Achter Lijn',
    category: 'nature',
    imagePos: { x: 0.11, y: 0.63 },
    description: 'Back line area',
  },
  {
    id: 'vuur-toren-strand',
    name: 'Vuur Toren Strand',
    category: 'water',
    imagePos: { x: 0.75, y: 0.66 },
    description: 'Fire tower beach',
  },

  // ── Services & Facilities ────────────────────────────────────────
  {
    id: 'ehbo',
    name: 'EHBO',
    category: 'medical',
    imagePos: { x: 0.18, y: 0.17 },
    description: 'First aid post',
  },
  {
    id: 'camping-winkel',
    name: 'Camping Winkel',
    category: 'facility',
    imagePos: { x: 0.24, y: 0.13 },
    description: 'Camping store – gas, supplies',
  },
  {
    id: 'dorps-tafel',
    name: 'Dorps Tafel',
    category: 'food',
    imagePos: { x: 0.38, y: 0.22 },
    description: 'Village table – communal dining',
  },
  {
    id: 'hourtentien-1',
    name: 'Hourtentien',
    category: 'facility',
    imagePos: { x: 0.74, y: 0.28 },
    description: 'Showers & toilets',
  },
  {
    id: 'hourtentien-2',
    name: 'Hourtentien',
    category: 'facility',
    imagePos: { x: 0.78, y: 0.43 },
    description: 'Showers & toilets',
  },
  {
    id: 'hourtentien-3',
    name: 'Hourtentien',
    category: 'facility',
    imagePos: { x: 0.69, y: 0.60 },
    description: 'Showers & toilets',
  },

  // ── Camping ──────────────────────────────────────────────────────
  {
    id: 'camping-1',
    name: 'Camping 1',
    category: 'camping',
    imagePos: { x: 0.47, y: 0.18 },
    description: 'Camping zone 1',
  },
  {
    id: 'camping-2',
    name: 'Camping 2',
    category: 'camping',
    imagePos: { x: 0.72, y: 0.37 },
    description: 'Camping zone 2',
  },
  {
    id: 'camping-4',
    name: 'Camping 4',
    category: 'camping',
    imagePos: { x: 0.21, y: 0.73 },
    description: 'Camping zone 4',
  },
  {
    id: 'camping-5',
    name: 'Camping 5',
    category: 'camping',
    imagePos: { x: 0.37, y: 0.78 },
    description: 'Camping zone 5',
  },
];

export const POI_CATEGORY_COLORS: Record<string, string> = {
  stage: '#FF6B35',
  camping: '#4CAF50',
  food: '#FFB300',
  medical: '#F44336',
  water: '#29B6F6',
  nature: '#66BB6A',
  activity: '#AB47BC',
  facility: '#78909C',
  entrance: '#FFFFFF',
};

export const POI_CATEGORY_ICONS: Record<string, string> = {
  stage: 'musical-notes',
  camping: 'bonfire',
  food: 'restaurant',
  medical: 'medkit',
  water: 'water',
  nature: 'leaf',
  activity: 'bicycle',
  facility: 'water-outline',
  entrance: 'enter',
};

File: composeApp/src/commonMain/kotlin/com/wildeburg/maps/data/POIs.kt
package com.wildeburg.maps.data

import androidx.compose.ui.graphics.Color

val POIS = listOf(
    // ── Stages ───────────────────────────────────────────────────────────────
    POI("wildlive", "Wildlive", POICategory.STAGE,
        GpsCoords(52.680115, 5.880332), "Main stage — live music"),
    POI("radio-koperen-hond", "Radio de Koperen Hond", POICategory.STAGE,
        GpsCoords(52.681806, 5.880589), "Stage & radio broadcast"),
    POI("bud-kas", "Bud Kas", POICategory.STAGE,
        GpsCoords(52.682528, 5.880851), "Bar & stage"),
    POI("lodge", "LODGE", POICategory.STAGE,
        GpsCoords(52.682968, 5.881256), "Stage"),
    POI("strand", "Strand", POICategory.STAGE,
        GpsCoords(52.680824, 5.881047), "Beach on the lake"),
    POI("de-baan", "De Baan", POICategory.STAGE,
        GpsCoords(52.680678, 5.882156), "Activity track"),
    POI("bamboe-bos", "Bamboe Bos", POICategory.STAGE,
        GpsCoords(52.682438, 5.881688), "Bamboo forest"),
    POI("helling", "Helling", POICategory.STAGE,
        GpsCoords(52.681479, 5.883310), "The slope"),
    POI("duin-pan", "Duin/Pan", POICategory.STAGE,
        GpsCoords(52.682970, 5.882118), "Dune & pan area"),
    POI("achter-tuin", "Achter Tuin", POICategory.STAGE,
        GpsCoords(52.682208, 5.885464), "Back line area"),
    POI("eiland", "Eiland", POICategory.STAGE,
        GpsCoords(52.683679, 5.883569)),

    // ── Areas ────────────────────────────────────────────────────────────────
    POI("de-spot", "De Spot", POICategory.AREA,
        GpsCoords(52.679827, 5.878452), "Electronic & DJ stage"),
    POI("vuur-toren-strand", "Vuur Toren Strand", POICategory.AREA,
        GpsCoords(52.684752, 5.884237), "Fire tower beach"),
    POI("bamboe-bios", "Bamboe Bios", POICategory.AREA,
        GpsCoords(52.683301, 5.884510), "Outdoor cinema in the bamboo"),

    // ── Services ─────────────────────────────────────────────────────────────
    POI("ehbo", "EHBO", POICategory.SERVICE,
        GpsCoords(52.679597, 5.879487), "First aid post"),
    POI("camping-winkel", "Camping Winkel", POICategory.SERVICE,
        GpsCoords(52.679685, 5.877996), "Camping store — gas, supplies"),

    // ── Camping ──────────────────────────────────────────────────────────────
    POI("camping-1", "Camping 1", POICategory.CAMPING,
        GpsCoords(52.680517, 5.878141), "Camping zone 1"),
    POI("camping-2", "Camping 2", POICategory.CAMPING,
        GpsCoords(52.683609, 5.878030), "Camping zone 2"),
    POI("camping-3", "Camping 3", POICategory.CAMPING,
        GpsCoords(52.685295, 5.880089), "Camping zone 3"),
    POI("camping-3-5", "Camping 3.5", POICategory.CAMPING,
        GpsCoords(52.685800, 5.883023), "Camping zone 3.5"),
    POI("camping-4", "Camping 4", POICategory.CAMPING,
        GpsCoords(52.683310, 5.885849), "Camping zone 4"),

    // ── Easter Eggs ──────────────────────────────────────────────────────────
    POI("shrub-1", "Shrub 1", POICategory.EASTER_EGG,
        GpsCoords(52.681756, 5.881749)),
    POI("the-shell", "The Shell", POICategory.EASTER_EGG,
        GpsCoords(52.681176, 5.880945)),
    POI("the-lost-condom", "The Lost Condom", POICategory.EASTER_EGG,
        GpsCoords(52.681418, 5.880753)),
    POI("home-base", "Home Base", POICategory.EASTER_EGG,
        GpsCoords(52.683410, 5.885377)),
    POI("showers", "Showers", POICategory.EASTER_EGG,
        GpsCoords(52.685214, 5.884922)),
    POI("chinese-wraps", "Chinese Wraps", POICategory.EASTER_EGG,
        GpsCoords(52.679532, 5.879131)),
    POI("erwins-lighter", "Erwin's Lighter", POICategory.EASTER_EGG,
        GpsCoords(52.681106, 5.882725)),
    POI("parking-for-legends", "Parking for Legends", POICategory.EASTER_EGG,
        GpsCoords(52.673824, 5.882195)),
    POI("drugs-in-the-bread", "Drugs in the Bread", POICategory.EASTER_EGG,
        GpsCoords(52.687407, 5.868129)),
)

fun poiColor(category: POICategory) = when (category) {
    POICategory.STAGE      -> Color(0xFFFF6B35)
    POICategory.AREA       -> Color(0xFFAB47BC)
    POICategory.SERVICE    -> Color(0xFF78909C)
    POICategory.CAMPING    -> Color(0xFF4CAF50)
    POICategory.EASTER_EGG -> Color(0xFFFFD54F)
}

fun poiEmoji(category: POICategory) = when (category) {
    POICategory.STAGE      -> "🎵"
    POICategory.AREA       -> "🎪"
    POICategory.SERVICE    -> "ℹ️"
    POICategory.CAMPING    -> "⛺"
    POICategory.EASTER_EGG -> "🥚"
}

fun categoryDisplayName(category: POICategory) = when (category) {
    POICategory.STAGE      -> "Stages"
    POICategory.AREA       -> "Areas"
    POICategory.SERVICE    -> "Services"
    POICategory.CAMPING    -> "Camping"
    POICategory.EASTER_EGG -> "Easter Eggs"
}
